tests/pytype_test.py

```diff
@@ -11,61 +11,68 @@
 """
 
 import argparse
-import collections
 import itertools
 import os
-from pytype import config
-from pytype import io
 import re
 import subprocess
-import sys
 import traceback
+from typing import List, Match, Optional, Sequence, Tuple
 
-parser = argparse.ArgumentParser(description='Pytype/typeshed tests.')
-parser.add_argument('-n', '--dry-run', action='store_true', default=False,
-                    help='Don\'t actually run tests')
-# Default to '' so that symlinking typeshed subdirs in cwd will work.
-parser.add_argument('--typeshed-location', type=str, default='',
-                    help='Path to typeshed installation.')
-# Set to true to print a stack trace every time an exception is thrown.
-parser.add_argument('--print-stderr', action='store_true', default=False,
-                    help='Print stderr every time an error is encountered.')
-# We need to invoke python2.7 and 3.6.
-parser.add_argument('--python27-exe', type=str, default='python2.7',
-                    help='Path to a python 2.7 interpreter.')
-parser.add_argument('--python36-exe', type=str, default='python3.6',
-                    help='Path to a python 3.6 interpreter.')
+from pytype import config as pytype_config, io as pytype_io
 
+TYPESHED_SUBDIRS = ["stdlib", "third_party"]
 
-TYPESHED_SUBDIRS = ['stdlib', 'third_party']
 
-
-TYPESHED_HOME = 'TYPESHED_HOME'
+TYPESHED_HOME = "TYPESHED_HOME"
 UNSET = object()  # marker for tracking the TYPESHED_HOME environment variable
 
 
-def main():
-    args = parser.parse_args()
-    code = pytype_test(args)
-    sys.exit(code)
-
-
-class PathMatcher(object):
-    def __init__(self, patterns):
-        if patterns:
-            self.matcher = re.compile('(%s)$' % '|'.join(patterns))
-        else:
-            self.matcher = None
-
-    def search(self, path):
+def main() -> None:
+    args = create_parser().parse_args()
+    typeshed_location = args.typeshed_location or os.getcwd()
+    subdir_paths = [os.path.join(typeshed_location, d) for d in TYPESHED_SUBDIRS]
+    check_subdirs_discoverable(subdir_paths)
+    check_python_exes_runnable(python27_exe_arg=args.python27_exe, python36_exe_arg=args.python36_exe)
+    if args.dry_run:
```

Comment:
The point of --dry-run is to run through all the code of pytype_test without actually calling into pytype itself for parsing (since that part is slow), so returning immediately defeats the purpose.

Reply:
Got it, will restore the original intent. Thanks!